flink-core/src/test/java/org/apache/flink/api/java/typeutils/TypeExtractorTest.java

```diff
@@ -1857,12 +1856,6 @@ public void testEitherHierarchy() {
 		Assert.assertEquals(expected, ti);
 	}
 
-	@Test(expected=InvalidTypesException.class)
-	public void testEitherFromObjectException() {
-		Either<String, Tuple1<Integer>> either = Either.Left("test");
-		TypeExtractor.getForObject(either);
```

Comment:
This test is necessary. The exception is not thrown anymore but the type information that is currently generated is invalid (the right type information is `null`, which should never happen).